### PR TITLE
feat: upgrade to ethers v6

### PR DESCRIPTION
**Pull request by dtbuchholz**
## Summary

Bumps ethers to v6—plus, OpenZepplin and Hardhat packages that need v6 as a peer dependency.

## Details

N/A

## How it was tested

Tested deployment scripts:
- Locally on hardhat node
- On Seplolia [here](https://sepolia.etherscan.io/address/0x94eCAF45504c15B6da0f30adB202c30004Ff047A#readProxyContract)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: scripts/deploy.ts [hunk 1]
```diff
@@ -29,20 +30,22 @@ async function main() {
...
   const Factory = await ethers.getContractFactory("TablelandTables");
-  const tables = await (
-    (await upgrades.deployProxy(Factory, [baseURI], {
-      kind: "uups",
-      timeout: pollTimeout,
-      pollingInterval: pollInterval,
-    })) as TablelandTables
-  ).deployed();
-  console.log("New proxy address:", tables.address);
+  // @ts-expect-error ignore `Conversion of type 'Contract'` error since
```
Comment (dtbuchholz) [inline scripts/deploy.ts:33]: i tried to get around this error but couldn't. the type that `deployProxy` _says_ it should return is an ethers `Contract`, but if you log it, it's a `BaseContract`—which is what `TablelandTables` extends. ignoring it resolved the issue.

File: scripts/deploy.ts [hunk 2]
```diff
@@ -29,20 +30,22 @@ async function main() {
...
+  const tables = await proxyDeploy.waitForDeployment();
+
+  console.log("New proxy address:", await tables.getAddress());
 
   // Check implementation
   // Note: We poll here because the impl won't be visible from the proxy until the next tipset on Filecoin.
   // Note: See https://docs.filecoin.io/smart-contracts/developing-contracts/best-practices/#consistently-generating-transaction-receipts.
   const startTime = Date.now();
-  while (!(await checkImpl(tables.address))) {
+  while (!(await checkImpl(await tables.getAddress()))) {
```
Comment (dtbuchholz) [inline scripts/deploy.ts:48]: the new way to get an addr is via method call

File: scripts/deploy.ts [hunk 3]
```diff
@@ -59,19 +62,28 @@ async function main() {
...
   );
   let receipt = await tx.wait();
-  const [, createEvent] = receipt.events ?? [];
+  const [, createEvent] = receipt?.logs.filter(isEventLog) ?? [];
   const tableId = createEvent.args!.tableId;
   console.log("Healthbot table created as:", `healthbot_${chainId}_${tableId}`);
 
   // Insert first row into health bot table
   const runStatement = `insert into healthbot_${chainId}_${tableId} values (1);`;
+  // Add 20% buffer to gas limit to avoid out-of-gas errors
```
Comment (dtbuchholz) [inline scripts/deploy.ts:71]: txs were failing when i tried to deploy to a live network, i.e., sepolia. the gas limit was hitting 99%+. this adds a buffer and fixed the issue.

File: scripts/utils.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import type { EventLog, Log } from "ethers";
```
Comment (dtbuchholz) [inline scripts/utils.ts:1]: the response from `receipt?.logs` (see above) is now a `Log` or `EventLog`, so we need to make sure we only get the `EventLog` in order to parse

**Review by sanderpick — APPROVED** [review]
very nice!

**Comment by dtbuchholz** [comment]
@sanderpick realized i didn't add tests here since i was rushing to test amoy support 😅 ...added. also, [hardhat dropped node 16](https://hardhat.org/hardhat-runner/docs/reference/stability-guarantees#node.js-versions-support) support, so i updated the gh action to use node 18 (and technically, they say they only support node LTS).

**Review by sanderpick — APPROVED** [review]

**Pull request merged**